test/core/channel/metrics_test.cc

```diff
@@ -146,12 +147,15 @@ class FakeStatsPlugin : public StatsPlugin {
   friend class FakeStatsPluginBuilder;
 
   explicit FakeStatsPlugin(
-      absl::AnyInvocable<bool(const ChannelScope& /*scope*/) const>
+      absl::AnyInvocable<bool(const StatsPlugin::ChannelScope& /*scope*/) const>
           channel_filter)
       : channel_filter_(std::move(channel_filter)) {
     GlobalInstrumentsRegistry::ForEach(
         [this](const GlobalInstrumentsRegistry::GlobalInstrumentDescriptor&
                    descriptor) {
+          if (!descriptor.enable_by_default) {
```

Comment:
Please also add a test that covers adding a metric that is not enabled by default.

Reply:
Done.